File: reference/spring-boot-java-maven-layered/src/main/java/com/example/demo/repository/WidgetRepository.java
package com.example.demo.repository;

import com.example.demo.domain.Widget;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface WidgetRepository extends JpaRepository<Widget, Long> {

    Optional<Widget> findByName(String name);

    boolean existsByName(String name);
}

File: reference/spring-boot-kotlin-gradle-layered/src/main/kotlin/com/example/demo/repository/WidgetRepository.kt
package com.example.demo.repository

import com.example.demo.domain.Widget
import org.springframework.data.jpa.repository.JpaRepository

interface WidgetRepository : JpaRepository<Widget, Long> {
    fun findByName(name: String): Widget?

    fun existsByName(name: String): Boolean
}
